### PR TITLE
[DW-47] - Analytics Service - Add a new method to get the report details by name

### PR DESCRIPTION
**Pull request by infad-kachancheri**
Fix for Issue#DW-47: Create a service that returns report details for a given report name as part of ng2-activiti-analytics component. There is no spec file for this service thats why skipped Unit Testcase.

https://issues.alfresco.com/jira/browse/DW-47

### CODE REVIEW

File: ng2-components/ng2-activiti-analytics/src/services/analytics.service.ts [hunk 1]
```diff
@@ -55,6 +55,22 @@ export class AnalyticsService {
...
+    /**
+     * Retrive Report by name
+     * @param reportName - string - The name of report
+     * @returns {Observable<any>}
+     */
+    getReportByName(reportName: string): Observable<any> {
+        return Observable.fromPromise(this.apiService.getInstance().activiti.reportApi.getReportList())
+            .map((response: any) => {
+
+                if (reportName) {
```
Comment (mauriziovitale) [inline ng2-components/ng2-activiti-analytics/src/services/analytics.service.ts]: We don't need this condition, the reportName is required as parameter

File: ng2-components/ng2-activiti-analytics/src/services/analytics.service.ts [hunk 2]
```diff
@@ -55,6 +55,22 @@ export class AnalyticsService {
...
+                    return response.find(p => p.name === reportName);
+                }
+                return null;
```
Comment (eromano) [inline ng2-components/ng2-activiti-analytics/src/services/analytics.service.ts]: never propagate null and provide only one exit point for method

File: ng2-components/ng2-activiti-analytics/src/services/analytics.service.ts (hunk 1 above)
Comment (infad-kachancheri) [inline ng2-components/ng2-activiti-analytics/src/services/analytics.service.ts]: these changes done.

File: ng2-components/ng2-activiti-analytics/src/services/analytics.service.ts (hunk 2 above)
Comment (infad-kachancheri) [inline ng2-components/ng2-activiti-analytics/src/services/analytics.service.ts]: these changes done.

**Comment by infad-kachancheri** [comment]
re-opened

**Pull request closed without merging**